### PR TITLE
TET-146 Support files with and without indices

### PR DESCRIPTION
**Pull request by GurgenAvagyan**
https://github.com/user-attachments/assets/7687d4e4-8a7c-4b1c-aa0f-d365a1f9e446

### CODE REVIEW

**Comment by HappyMary16** [comment]
Let's not add the same models without indices, you may leave just one small model for testing purposes in the project

File: src/main/java/org/cmps/tetrahedron/utils/DataReader.java [hunk 1]
```diff
@@ -19,15 +20,25 @@ public static Map<Integer, float[]> readVertices(File coordinatesTableFile) {
...
                     index = i++;
+                    startIndex = 0;
+                }
+
+                for (int j = startIndex; j < elements.length; j++) {
+                    vertex.add(Float.parseFloat(elements[j]));
                 }
 
-                coordinates.put(index, new float[]{fid.nextFloat(), fid.nextFloat(), fid.nextFloat()});
+                coordinates.put(index, convertListToArray(vertex));
```
Comment (HappyMary16) [inline src/main/java/org/cmps/tetrahedron/utils/DataReader.java:40]: Can't we store coordinates in the array when we read them? This conversion looks redundant
We know the length of the coordinates array, so we shouldn't have any issues with creating it.

**Review by HappyMary16 — CHANGES_REQUESTED** [review]

File: src/main/java/org/cmps/tetrahedron/utils/DataReader.java [hunk 2]
```diff
@@ -42,17 +52,22 @@ public static List<float[][]> readIndexesAndConvertToFaces(File indicesMatrix,
...
-            while (fid.hasNext()) {
-                // reading index of a tetrahedron
-                if (WITH_INDICES) {
-                    fid.nextInt();
+            while (fid.hasNextLine()) {
+                String[] elements = fid.nextLine().trim().split("\\s+");
+                float[] vertex1, vertex2, vertex3, vertex4;
+
+                if (elements.length == FACE_WITH_INDICES) {
+                    vertex1 = verticesCoordinates.get(Integer.parseInt(elements[1]));
```
Comment (HappyMary16) [inline src/main/java/org/cmps/tetrahedron/utils/DataReader.java:60]: are there any ways not to duplicate the code inside the if-else operator?

Comment (GurgenAvagyan) [reply src/main/java/org/cmps/tetrahedron/utils/DataReader.java:60]: I've tried, but so far I couldn't done it.

**Review by VadimST04 — APPROVED** [review]

**Pull request merged**